test/PerformanceHints/existential-any.swift

```diff
@@ -0,0 +1,345 @@
+// RUN: %empty-directory(%t)
+
+// Ensure ignored by-default
+// RUN: %target-swift-frontend -typecheck %s -diagnostic-style llvm -verify
+
+// Ensure enabled with '-Wwarning'
+// RUN: %target-swift-frontend -typecheck %s -diagnostic-style llvm -Wwarning PerformanceHints &> %t/output_warn.txt
+// RUN: cat %t/output_warn.txt | %FileCheck %s -check-prefix CHECK-WARN
+// RUN: %target-swift-frontend -typecheck %s -diagnostic-style llvm -Wwarning ExistentialAnyType &> %t/output_warn.txt
+// RUN: cat %t/output_warn.txt | %FileCheck %s -check-prefix CHECK-WARN
+
+// Ensure enabled with '-Werror' for the broad category and downgraded to warning for the subgroup with '-Wwarning'
+// RUN: %target-swift-frontend -typecheck %s -diagnostic-style llvm -Werror PerformanceHints -Wwarning ExistentialAnyType &> %t/output_warn.txt
+// RUN: cat %t/output_warn.txt | %FileCheck %s -check-prefix CHECK-WARN
+
+// Ensure enabled with '-Wwarning' for the broad category and downgraded to warning for the subgroup with '-Werror'
+// RUN: not %target-swift-frontend -typecheck %s -diagnostic-style llvm -Wwarning PerformanceHints -Werror ExistentialAnyType &> %t/output_err.txt
+// RUN: cat %t/output_err.txt | %FileCheck %s -check-prefix CHECK-ERR
+
+// Ensure escalated with '-Werror'
+// RUN: not %target-swift-frontend -typecheck %s -diagnostic-style llvm -Werror ExistentialAnyType &> %t/output_err.txt
+// RUN: cat %t/output_err.txt | %FileCheck %s -check-prefix CHECK-ERR
+
+// CHECK-ERR: error: Performance: 'AnyAnimal' aliases existential any type, leading to heap allocation, reference counting, and dynamic dispatch. Consider using generic constraints or concrete types instead. [#ExistentialAnyType]
```

Comment:
There's so many of these, I think it would be a lot cleaner if we just invoked the compilers here with `-verify` and checked the diagnostics inline.